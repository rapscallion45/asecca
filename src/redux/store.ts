import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import logger from 'redux-logger';
import userPermissionReducer from './slices/userPermissionSlice';
import costsConfigReducer from './slices/costsConfigSlice';
import notificationsReducer from './slices/notificationsSlice';
import themeReducer from './slices/themeSlice';
import { collectionsKanbanSlice } from './slices/kanbanSlice';

/**
 * Configuration of the redux store to manage the global application state
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @see See [more info on Redux Store object](https://redux.js.org/api/store)
 *
 * @constant
 * @type {ToolkitStore<Object>}
 */
const store = configureStore({
  reducer: {
    userPermission: userPermissionReducer,
    costsConfig: costsConfigReducer,
    notifications: notificationsReducer,
    theme: themeReducer,
    collectionsKanban: collectionsKanbanSlice.reducer,
  },
  // @ts-ignore
  middleware:
    process.env.NODE_ENV === 'development'
      ? (getDefaultMiddleware) => getDefaultMiddleware().concat(logger)
      : (getDefaultMiddleware) => getDefaultMiddleware(),
});

/**
 * Shorthand application store dispatch type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppDispatch - application specific store dispatch type
 */
export type AppDispatch = typeof store.dispatch;

/**
 * Shorthand application state type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppState - application specific state object shape
 */
export type AppState = ReturnType<typeof store.getState>;

/**
 * Shorthand application Thunk Action type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppThunk - application specific Thunk action definition
 */
export type AppThunk = ThunkAction<void, AppState, unknown, Action<string>>;

export default store;
