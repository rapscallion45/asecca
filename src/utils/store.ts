/* redux store configuration placeholder */
const useStore = () => {};
export default useStore;
