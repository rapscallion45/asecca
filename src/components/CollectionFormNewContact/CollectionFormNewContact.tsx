import { FC, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useFormik } from 'formik';
import * as Yup from 'yup';
import { AppDispatch } from '@/redux/store';
import {
  Box,
  Button,
  TextField,
  Grid,
  Card,
  CardHeader,
  CardContent,
} from '@mui/material';
import { LoadingButton } from '@mui/lab';
import { addNotification } from '@/redux/slices/notificationsSlice';
import {
  IContactData,
  ICollectionFormNewContactSaveDataPayload,
} from '@/lib/api/api-types';
import collectionFormService from '@/services/forms/collectionFormService';

/**
 * Collection Form New Contact Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.17
 *
 * @typedef ICollectionFormNewContactProps
 * @prop {string} collectionId - ID string of Collection for data API call
 */
interface ICollectionFormNewContactProps {
  collectionId: string;
}

/**
 * Collection Form New Contact
 *
 * Presents the Collection Form New Asset modal to the user, populated with data
 * fetched from API: /api/collection/asset_category/api/facilities
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.17
 *
 * @component
 * @param {ICollectionFormNewContactProps} props - component props
 * @returns {FC} - collection form new asset modal functional component
 */
const CollectionFormNewContact: FC<ICollectionFormNewContactProps> = (
  props
) => {
  const { collectionId } = props;
  const dispatch = useDispatch<AppDispatch>();

  /**
   * Local state data for new contact saving flag
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.17
   *
   * @constant
   */
  const [saving, setSaving] = useState<boolean>(false);

  /**
   * Yup input validation configuration for form data
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.17
   *
   * @constant
   */
  const validationSchema = Yup.object().shape({
    name: Yup.object().shape({
      prefix: Yup.string()
        .required('Required')
        .max(6, 'Contact prefix must not exceed 6 characters'),
      first: Yup.string()
        .required('Contact first name is required')
        .max(60, 'Contact first name must not exceed 60 characters'),
      last: Yup.string()
        .required('Contact last name is required')
        .max(60, 'Contact last name must not exceed 60 characters'),
    }),
    email: Yup.string()
      .email('Please enter a valid email address')
      .required('Contact email is required'),
    mobile: Yup.string().matches(
      /^[0-9+]+$/,
      'Mobile number must only contain digits'
    ),
    landline: Yup.string().matches(
      /^[0-9+]+$/,
      'Landline must only contain digits'
    ),
  });

  /**
   * Data submission handler for form
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.17
   *
   * @method
   * @param {ICollectionFormNewContactSaveDataPayload} payload - data to be submitted
   */
  const handleSubmit = async (
    payload: ICollectionFormNewContactSaveDataPayload
  ) => {
    /* indicate save in progress */
    setSaving(true);

    /* call API */
    const res = await collectionFormService.createNewContact(payload);

    /* clear save in progress flag */
    setSaving(false);

    /* add a notification and reject if bad response from server */
    if (res.status !== 200) {
      dispatch(
        addNotification({
          message: `Failed to save New Contact: ${res.statusText}`,
          variant: 'error',
        })
      );
      return;
    }

    /* indicate successful save of New Contact */
    dispatch(
      addNotification({
        message: `Successfully saved New Contact: ${res.statusText}`,
        variant: 'success',
      })
    );
  };

  /**
   * Formik configuration for form data
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.17
   *
   * @constant
   */
  const formik = useFormik<IContactData>({
    initialValues: {
      name: {
        prefix: '',
        first: '',
        last: '',
      },
      email: '',
      job_title: '',
      landline: '',
      mobile: '',
    },
    validationSchema,
    onSubmit: (data: IContactData) => {
      handleSubmit({ collectionId, ...data });
    },
  });

  return (
    <Card>
      <CardHeader title="New Contact" />
      <CardContent sx={{ pt: 0 }}>
        <form onSubmit={formik.handleSubmit}>
          <Grid container spacing={2}>
            <Grid item xs={12} sm={2}>
              <TextField
                id="prefix"
                name="name.prefix"
                type="text"
                label="Prefix"
                value={formik.values.name.prefix}
                onChange={formik.handleChange}
                error={
                  formik.touched.name?.prefix &&
                  Boolean(formik.errors.name?.prefix)
                }
                helperText={
                  formik.touched.name?.prefix && formik.errors.name?.prefix
                }
                placeholder="New Contact Name Prefix..."
                fullWidth
                color="secondary"
              />
            </Grid>
            <Grid item xs={12} sm={5}>
              <TextField
                id="first"
                name="name.first"
                type="text"
                label="First Name"
                value={formik.values.name.first}
                onChange={formik.handleChange}
                error={
                  formik.touched.name?.first &&
                  Boolean(formik.errors.name?.first)
                }
                helperText={
                  formik.touched.name?.first && formik.errors.name?.first
                }
                placeholder="New Contact First Name..."
                fullWidth
                color="secondary"
              />
            </Grid>
            <Grid item xs={12} sm={5}>
              <TextField
                id="last"
                name="name.last"
                type="text"
                label="Last Name"
                value={formik.values.name.last}
                onChange={formik.handleChange}
                error={
                  formik.touched.name?.last && Boolean(formik.errors.name?.last)
                }
                helperText={
                  formik.touched.name?.last && formik.errors.name?.last
                }
                placeholder="New Contact Last Name..."
                fullWidth
                color="secondary"
              />
            </Grid>
            <Grid item xs={12} sm={12}>
              <TextField
                id="email"
                name="email"
                type="email"
                label="Email"
                value={formik.values.email}
                onChange={formik.handleChange}
                error={formik.touched.email && Boolean(formik.errors.email)}
                helperText={formik.touched.email && formik.errors.email}
                placeholder="New Contact Email..."
                fullWidth
                color="secondary"
              />
            </Grid>
            <Grid item xs={12} sm={12}>
              <TextField
                id="job_title"
                name="job_title"
                type="text"
                label="Job Title"
                value={formik.values.job_title}
                onChange={formik.handleChange}
                error={
                  formik.touched.job_title && Boolean(formik.errors.job_title)
                }
                helperText={formik.touched.job_title && formik.errors.job_title}
                placeholder="New Contact Job Title..."
                fullWidth
                color="secondary"
              />
            </Grid>
            <Grid item xs={12} sm={6}>
              <TextField
                id="landline"
                name="landline"
                type="text"
                label="Landline"
                value={formik.values.landline}
                onChange={formik.handleChange}
                error={
                  formik.touched.landline && Boolean(formik.errors.landline)
                }
                helperText={formik.touched.landline && formik.errors.landline}
                placeholder="New Contact Landline..."
                fullWidth
                color="secondary"
              />
            </Grid>
            <Grid item xs={12} sm={6}>
              <TextField
                id="mobile"
                name="mobile"
                type="text"
                label="Mobile"
                value={formik.values.mobile}
                onChange={formik.handleChange}
                error={formik.touched.mobile && Boolean(formik.errors.mobile)}
                helperText={formik.touched.mobile && formik.errors.mobile}
                placeholder="New Contact Mobile..."
                fullWidth
                color="secondary"
              />
            </Grid>
          </Grid>
          <Box
            sx={{
              marginTop: 3,
              display: 'flex',
              flexDirection: 'row',
              alignItems: 'center',
              justifyContent: 'left',
            }}
          >
            <Button
              color="secondary"
              variant="outlined"
              onClick={formik.handleReset}
              disabled={saving}
              sx={{ backgroundColor: 'common.white' }}
            >
              Cancel
            </Button>
            <LoadingButton
              type="submit"
              color="secondary"
              variant="contained"
              disabled={saving}
              loading={saving}
              sx={{ ml: 2 }}
            >
              Create
            </LoadingButton>
          </Box>
        </form>
      </CardContent>
    </Card>
  );
};

export default CollectionFormNewContact;
