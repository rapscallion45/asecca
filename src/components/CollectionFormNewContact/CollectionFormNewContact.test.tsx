import React from 'react';
import { Provider } from 'react-redux';
import {
  fireEvent,
  render,
  screen,
  waitFor,
  act,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import store from '@/redux/store';
import CollectionFormNewContact from './CollectionFormNewContact';

/* default test query ID */
const query: string = '66135000001760012';

/**
 * Collection Form New Contact Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.22
 */
describe('Collection Form New Contact', () => {
  it('Renders correctly', async () => {
    /** Arrange */
    /** perform snapshot test */
    const tree = renderer
      .create(
        <Provider store={store}>
          <CollectionFormNewContact collectionId={query} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Form Fields', () => {
    it('Should render all required New Contact data fields', async () => {
      /** Arrange */
      /** Act - render the test components */
      render(
        <Provider store={store}>
          <CollectionFormNewContact collectionId={query} />
        </Provider>
      );

      /** Assert - ensure all fields are present */
      await waitFor(() => {
        expect(screen.getByLabelText('Prefix')).toBeInTheDocument();
      });
      expect(screen.getByLabelText('First Name')).toBeInTheDocument();
      expect(screen.getByLabelText('Last Name')).toBeInTheDocument();
      expect(screen.getByLabelText('Email')).toBeInTheDocument();
      expect(screen.getByLabelText('Job Title')).toBeInTheDocument();
      expect(screen.getByLabelText('Landline')).toBeInTheDocument();
      expect(screen.getByLabelText('Mobile')).toBeInTheDocument();
      expect(screen.getByLabelText('Sync To CRM')).toBeInTheDocument();
    });
  });

  describe('Form Callbacks', () => {
    beforeAll(() => jest.spyOn(window, 'fetch'));

    it('Should call API with inputted values when values updated and save button clicked', async () => {
      /** Arrange */
      const testPrefix = 'Mr';
      const testFirstName = 'John';
      const testLastName = 'Smith';
      const testEmail = 'john.smith@test.com';
      const testJobTitle = 'Software Developer';
      const testLandline = '03456123456';
      const testMobile = '07793388668';

      /** Act - render the test components */
      render(
        <Provider store={store}>
          <CollectionFormNewContact collectionId={query} />
        </Provider>
      );
      await waitFor(() => {
        expect(screen.getByLabelText('Prefix')).toBeInTheDocument();
      });
      /** Act - update field values and click Save button */
      act(() => {
        fireEvent(
          screen.getByLabelText('Prefix'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
        fireEvent.change(
          screen.getByPlaceholderText('New Contact Name Prefix...'),
          {
            target: { value: testPrefix },
          }
        );
      });
      act(() => {
        fireEvent(
          screen.getByLabelText('First Name'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
        fireEvent.change(
          screen.getByPlaceholderText('New Contact First Name...'),
          {
            target: { value: testFirstName },
          }
        );
      });
      act(() => {
        fireEvent(
          screen.getByLabelText('Last Name'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
        fireEvent.change(
          screen.getByPlaceholderText('New Contact Last Name...'),
          {
            target: { value: testLastName },
          }
        );
      });
      act(() => {
        fireEvent(
          screen.getByLabelText('Email'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
        fireEvent.change(screen.getByPlaceholderText('New Contact Email...'), {
          target: { value: testEmail },
        });
      });
      act(() => {
        fireEvent(
          screen.getByLabelText('Job Title'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
        fireEvent.change(
          screen.getByPlaceholderText('New Contact Job Title...'),
          {
            target: { value: testJobTitle },
          }
        );
      });
      act(() => {
        fireEvent(
          screen.getByLabelText('Landline'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
        fireEvent.change(
          screen.getByPlaceholderText('New Contact Landline...'),
          {
            target: { value: testLandline },
          }
        );
      });
      act(() => {
        fireEvent(
          screen.getByLabelText('Mobile'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
        fireEvent.change(screen.getByPlaceholderText('New Contact Mobile...'), {
          target: { value: testMobile },
        });
      });
      /** Act - click Create button */
      act(() => {
        fireEvent(
          screen.getByText('Create'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
      });
      /** Assert - save API to have been called with data from state */
      await waitFor(() => {
        expect(window.fetch).toHaveBeenCalledTimes(1);
      });
      expect(window.fetch).toHaveBeenCalledWith(
        '/api/collection/contact/api/create',
        expect.objectContaining({
          method: 'POST',
          headers: { 'Content-Type': 'application/json' },
          body: JSON.stringify({
            collectionId: query,
            name: {
              prefix: testPrefix,
              first: testFirstName,
              last: testLastName,
            },
            email: testEmail,
            job_title: testJobTitle,
            landline: testLandline,
            mobile: testMobile,
            sync_to_crm: false,
          }),
        })
      );
    });
  });
});
