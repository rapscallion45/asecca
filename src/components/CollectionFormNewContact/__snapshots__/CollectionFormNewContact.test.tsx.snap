// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Form New Contact Renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
      >
        New Contact
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root css-1hizqsf-MuiCardContent-root"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="prefix"
              id="prefix-label"
            >
              Prefix
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="prefix"
                name="name.prefix"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="New Contact Name Prefix..."
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Prefix
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 css-11lxi55-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="first"
              id="first-label"
            >
              First Name
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="first"
                name="name.first"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="New Contact First Name..."
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    First Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 css-11lxi55-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="last"
              id="last-label"
            >
              Last Name
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="last"
                name="name.last"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="New Contact Last Name..."
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Last Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="email"
              id="email-label"
            >
              Email
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="email"
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="New Contact Email..."
                required={false}
                type="email"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="job_title"
              id="job_title-label"
            >
              Job Title
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="job_title"
                name="job_title"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="New Contact Job Title..."
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Job Title
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="landline"
              id="landline-label"
            >
              Landline
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="landline"
                name="landline"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="New Contact Landline..."
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Landline
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="mobile"
              id="mobile-label"
            >
              Mobile
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="mobile"
                name="mobile"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="New Contact Mobile..."
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Mobile
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
            id="sync_to_crm"
          >
            <span
              className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-root MuiCheckbox-colorSecondary css-1sg8fd1-MuiButtonBase-root-MuiCheckbox-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                name="sync_to_crm"
                onChange={[Function]}
                required={false}
                type="checkbox"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Sync To CRM
            </span>
          </label>
        </div>
      </div>
      <div
        className="MuiBox-root css-yiadgv"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1ex1hsz-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Cancel
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1fddvdp-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          disabled={false}
          id=":r7:"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Create
        </button>
      </div>
    </form>
  </div>
</div>
`;
